graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/directives/LinkDirectiveProcessor.java

```diff
@@ -95,10 +100,19 @@ private static Stream<SDLNamedDefinition> loadDefinitions(Directive linkDirectiv
   }
 
   private static int parseFederationVersion(String specLink) {
-    final String versionString = specLink.substring(specLink.length() - 3);
+    final Matcher matcher = LINK_FED_VERSION_PATTERN.matcher(specLink);
+    if (!matcher.find() || matcher.groupCount() != 2) {
+      throw new UnsupportedFederationVersionException(specLink);
+    }
+
     try {
-      return Math.round(Float.parseFloat(versionString) * 10);
-    } catch (Exception e) {
+      int major = Integer.parseInt(matcher.group(1));
+      final String minorStr = matcher.group(2);
+
+      // Calculate: major * 10^(digits in minor) + minor
+      // Examples: v2.3 → 2*10 + 3 = 23, v2.12 → 2*100 + 12 = 212
+      return (int) (major * Math.pow(10, minorStr.length()) + Integer.parseInt(minorStr));
```

Comment:
since we will only (realistically) do xx.yy versioning (i.e. no triple digit minors) I think we can simplify it

```suggestion
      int major = Integer.parseInt(matcher.group(1));
      int minor = Integer.parseInt(matcher.group(2));

      // Calculate: major * 100 + minor
      // Examples: v2.3 → 2*100 + 3 = 203, v2.12 → 2*100 + 12 = 212
      return (int) (major * 100 + minor);
```